LanguageWeaverProvider/Services/CloudService.cs

```diff
@@ -0,0 +1,229 @@
+				var dictionariesJson = JObject.Parse(responseContent)["dictionaries"].ToString();
+				var dictionariesOutput = JsonConvert.DeserializeObject<List<PairDictionary>>(dictionariesJson);
+
+				return dictionariesOutput;
+			}
+			catch (Exception ex)
+			{
+				throw ex;
+			}
+		}
+
+		public static async Task<Xliff> Translate(AccessToken accessToken, PairMapping mappedPair, Xliff sourceXliff)
+		{
+			try
+			{
+				var translationRequestResponse = await SendTranslationRequest(accessToken, mappedPair, sourceXliff);
+				var translationRequest = JsonConvert.DeserializeObject<TranslationRequestResponse>(translationRequestResponse);
+
+				var translationStatusReponse = await GetTranslationStatus(accessToken, translationRequest.RequestId);
+				var translationStatus = JsonConvert.DeserializeObject<TranslationStatus>(translationStatusReponse);
+				while (translationStatus.Status != "DONE")
+				{
+					if (translationStatus.Status == "FAILED")
+					{
+						return null;
+					}
+
+					Thread.Sleep(500);
+					translationStatusReponse = await GetTranslationStatus(accessToken, translationRequest.RequestId);
+					translationStatus = JsonConvert.DeserializeObject<TranslationStatus>(translationStatusReponse);
+				}
+
+				var translationResponse = await GetTranslation(accessToken, translationRequest.RequestId);
+				var translation = JsonConvert.DeserializeObject<TranslationResponse>(translationResponse);
+
+				var translatedSegment = translation.Translation.First();
+				var translatedXliffSegment = Converter.ParseXliffString(translatedSegment);
+				return translatedXliffSegment;
+			}
+			catch (Exception ex)
+			{
+				throw ex;
+			}
+		}
+
+		private static async Task<string> GetTranslationStatus(AccessToken accessToken, string requestId)
+		{
+			var httpClient = new HttpClient();
+			httpClient.DefaultRequestHeaders.Add("Authorization", $"{accessToken.TokenType} {accessToken.Token}");
+
+			var endpoint = $"https://api.languageweaver.com/v4/mt/translations/async/{requestId}?includeProgressInfo=true";
+			var response = await httpClient.GetStringAsync(endpoint);
+			return response;
+		}
+
+		private static async Task<string> SendTranslationRequest(AccessToken accessToken, PairMapping mappedPair, Xliff sourceXliff)
+		{
+			const string InputFormat = "xliff";
+			const string QualityEstimationAbbreviation = "qe";
+
+			var sourceXliffText = sourceXliff.ToString();
+			var linguisticOptionsDictionary = mappedPair.LinguisticOptions?.ToDictionary(lo => lo.Id, lo => lo.SelectedValue);
+
+			var translationRequestModel = new TranslationRequest
+			{
+				SourceLanguageId = mappedPair.SourceCode,
+				TargetLanguageId = mappedPair.TargetCode,
+				Input = new[] { sourceXliff.ToString() },
+				Model = mappedPair.SelectedModel.Model,
+				InputFormat = InputFormat,
+				Dictionaries = new object[] { },
+				QualityEstimation = mappedPair.SelectedModel.Model.ToLower().Contains(QualityEstimationAbbreviation) ? 1 : 0,
```

Comment:
Info from Arnaud Simon (Language Weaver): 
"Had a sync with engineering: the legacy "qualityEstimation" parameter of the translation API endpoint has no more effect since qualityEstimation was moved to a linguistic option. We are going to retire it. Your integration code should then focus on the use of Linguistic options for QE."